go/flagutil/enum_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewStringEnumCaseSensitive(t *testing.T) {
+	enum := NewStringEnum("fruits enum", "Mango", []string{"apple", "Mango"})
```

Comment:
Why not fold these functions in as test cases in the `TestStringEnum` test function?